File: scripts/fix-giveaways-jsx.py
from pathlib import Path

p = Path(__file__).resolve().parents[1] / "src/app/giveaways/page.tsx"
s = p.read_text(encoding="utf-8")

WRONG = "</" + "motion.div>"
DIV = "</" + "div>"
MOTION = "</" + "motion.div>"

# Referral header: <motion.div> opened line 222, wrong close
s = s.replace(
    "+150 pts each signup</span>\n        " + WRONG + "\n        <div className=\"flex gap-2\">",
    "+150 pts each signup</span>\n        " + DIV + "\n        <div className=\"flex gap-2\">",
)

# Icon box: <motion.div> line 278, wrong close line 280
s = s.replace(
    'done ? "text-emerald-400" : "text-brand")} />\n        ' + WRONG + "\n                <div className=\"flex-1",
    'done ? "text-emerald-400" : "text-brand")} />\n                ' + DIV + "\n                <div className=\"flex-1",
)

# Earn row: <motion.div> line 271, wrong </div> close line 309
s = s.replace(
    "                )}\n              </motion.div>\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
    "                )}\n              " + MOTION + "\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
)

# If earn row still has plain div close, fix it
s = s.replace(
    "                )}\n              </motion.div>\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
    "                )}\n              " + MOTION + "\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
)

# Earn row fix - use DIV variable for wrong close
s = s.replace(
    "                )}\n              " + DIV + "\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
    "                )}\n              " + MOTION + "\n            );\n          })}\n        </div>\n      )}\n\n      {tab === \"leaderboard\"",
)

p.write_text(s, encoding="utf-8")
print("fixed")

File: scripts/fix-giveaways-jsx.mjs
import fs from "fs";

const p = "src/app/giveaways/page.tsx";
let s = fs.readFileSync(p, "utf8");

// Referral header: div open, motion close -> div close
s = s.replace(
  `+150 pts each signup</span>
        </motion.div>
        <motion.div className="flex gap-2">`,
  `+150 pts each signup</span>
        </motion.div>
        <motion.div className="flex gap-2">`
);

// Fix referral header close
s = s.replace(
  "+150 pts each signup</span>\n        </motion.div>\n        <div className=\"flex gap-2\">",
  "+150 pts each signup</span>\n        </div>\n        <div className=\"flex gap-2\">"
);

// Icon box: div open, motion close
s = s.replace(
  /(<action\.icon className=\{cn\("w-5 h-5", done \? "text-emerald-400" : "text-brand"\)\} \/>)\n                <\/motion\.div>/,
  "$1\n                </div>"
);

// Earn row: motion open, div close
s = s.replace(
  /(\{claiming === action\.action \? <Loader2 className="w-3 h-3 animate-spin" \/> : action\.cta\}\n                  <\/button>\n                \)\}\n              )<\/div>/,
  `$1</motion.div>`
);

fs.writeFileSync(p, s);
console.log("fixed");
